### PR TITLE
fix: cog can still be MISSING

### PR DESCRIPTION
**Pull request by ChunkLightTuna**
## Summary

https://github.com/Pycord-Development/pycord/issues/2303 partially changed checks of `MISSING` to `None` with the assumption that cog would never be `MISSING`. Cog is `MISSING` for commands instantiated from `@bot.slash_command()`. 

## Information

- [x] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed).
- [ ] This PR is **not** a code change (e.g. documentation, README, typehinting,
      examples, ...).

## Checklist

- [x] I have searched the open pull requests for duplicates.
- [x] If code changes were made then they have been tested.
  - [ ] I have updated the documentation to reflect the changes.
- [ ] If `type: ignore` comments were used, a comment is also left explaining why.
- [x] I have updated the changelog to include these changes.

### CODE REVIEW

**Comment by Dorukyum** [comment]
I have missed a part of code, yes, but your changes affect the logic too. Will review shortly.

File: discord/bot.py [hunk 1]
```diff
@@ -267,9 +264,9 @@ def _check_command(cmd: ApplicationCommand, match: Mapping[str, Any]) -> bool:
                             for data in match["options"]
                             if data["name"] == subcommand.name
                         ),
-                        MISSING,
+                        None,
                     )
-                    if match_ is not MISSING and _check_command(subcommand, match_):
+                    if match_ and _check_command(subcommand, match_):
```
Comment (Dorukyum) [inline discord/bot.py:269]: I don't think this has anything to do with the cog attribute being MISSING, the MISSING value here is used for matching commands.

File: CHANGELOG.md [hunk 2]
```diff
@@ -207,6 +207,8 @@ These changes are available on the `master` branch, but have not yet been releas
   ([#2301](https://github.com/Pycord-Development/pycord/pull/2301))
 - Fixed `AttributeError` caused by `command.cog` being `MISSING`.
   ([#2303](https://github.com/Pycord-Development/pycord/issues/2303))
+- Fixed command options instantiated from @bot.slash_command().
+  ([#2313](https://github.com/Pycord-Development/pycord/issues/2313))
```
Comment (Dorukyum) [inline CHANGELOG.md:211]: I guess these belong together
```suggestion
- Fixed `AttributeError` caused by `command.cog` being `MISSING`.
  ([#2303](https://github.com/Pycord-Development/pycord/issues/2303) &
  [#2313](https://github.com/Pycord-Development/pycord/issues/2313))
```

File: discord/bot.py [hunk 3]
```diff
@@ -298,17 +295,15 @@ def _check_command(cmd: ApplicationCommand, match: Mapping[str, Any]) -> bool:
                         falsy_vals = (False, [])
                         for opt in value:
                             cmd_vals = (
-                                [val.get(opt, MISSING) for val in as_dict[check]]
+                                [val.get(opt) for val in as_dict[check]]
```
Comment (Dorukyum) [inline discord/bot.py:298]: Neither `as_dict` nor `to_check` will ever contain `cog` in this chunk of the code. I don't understand how this fixes the issue, MISSING was again used for matching commands.

**Review by Dorukyum — CHANGES_REQUESTED** [review]

File: discord/bot.py (hunk 3 above)
Comment (ChunkLightTuna) [reply discord/bot.py:298]: I’ll need to test this more. Earlier when I had *accidentally* inverted the logic, it fixed the issue with non-cog commands, but now when properly treating None as a drop in replacement for MISSING it's again not retrieving options.

**Comment by Dorukyum** [comment]
It seems your thesis was wrong and I'm fixing the actual issue in a different pull request. Thanks anyway.

**Pull request closed without merging**